sdk/identity/identity/src/credentials/deviceCodeCredential.ts

```diff
@@ -160,12 +160,16 @@ export class DeviceCodeCredential implements TokenCredential {
   ): Promise<AccessToken | null> {
     try {
       const deviceResponse = await this.msalClient.acquireTokenByDeviceCode(deviceCodeRequest);
-      const expiresOnTimestamp = deviceResponse.expiresOn.getTime();
-      logger.getToken.info(formatSuccess(scopes));
-      return {
-        expiresOnTimestamp,
-        token: deviceResponse.accessToken
-      };
+      if (deviceResponse && deviceResponse.expiresOn) {
+        const expiresOnTimestamp = deviceResponse.expiresOn.getTime();
+        logger.getToken.info(formatSuccess(scopes));
+        return {
+          expiresOnTimestamp,
+          token: deviceResponse.accessToken
+        };
+      } else {
+        throw new Error(`Device Authentication Error "Did not receive token with timestamp"`);
```

Comment:
This error is caught a few lines below and re-thrown. The final error message will look like the below with `Device Authentication Error" repeating.

> `Device Authentication Error "Device Authentication Error "Did not receive token with timestamp""

Reply:
Good catch